src/dz-5/hw5.ts

```diff
@@ -0,0 +1,52 @@
+abstract class Shape {
+    constructor(readonly color: string, readonly name: string) { }
+
+    abstract calculateArea(): number
+}
+
+class Circle extends Shape {
+    constructor(color: string, name: string, readonly radius: number) {
+        super(color, name);
+    }
+
+    calculateArea(): number {
+        return Math.PI * this.radius ** 2;
+    }
+}
+
+class Rectangle extends Shape {
+    constructor(color: string, name: string, readonly width: number, readonly height: number) {
+        super(color, name);
+    }
+
+    calculateArea(): number {
+        return this.width * this.height;
+    }
+
+    print(): void {
```

Comment:
Help me to avoid missing the "Print" method. You already know how to do this.

Reply:
add it to an abstract class and make it optional?

Reply:
Yes, you can add another abstract class for that. Or create an interface. The final decision is up to you.